src/Microsoft.Health.Fhir.Shared.Api/Modules/FeatureFlags/ConditionalCreates/ConditionalCreatePostConfigureOptions.cs

```diff
@@ -17,40 +16,32 @@ namespace Microsoft.Health.Fhir.Api.Modules.FeatureFlags.ConditionalCreates
 {
     public class ConditionalCreatePostConfigureOptions : IPostConfigureOptions<MvcOptions>
     {
-        private readonly FeatureConfiguration _features;
         private readonly IConfiguredConformanceProvider _configuredConformanceProvider;
 
         public ConditionalCreatePostConfigureOptions(
-            IOptions<FeatureConfiguration> features,
             IConfiguredConformanceProvider configuredConformanceProvider)
         {
-            EnsureArg.IsNotNull(features, nameof(features));
             EnsureArg.IsNotNull(configuredConformanceProvider, nameof(configuredConformanceProvider));
 
-            _features = features.Value;
             _configuredConformanceProvider = configuredConformanceProvider;
         }
 
         public void PostConfigure(string name, MvcOptions options)
         {
-            if (_features.SupportsConditionalCreate)
-            {
-                // Turns on conditional creates, even when not configured in the DefaultCapabilities.json
-                // If this is not enabled in the capability statement
-                // ConditionalCreateResourceRequest will fail with a MethodNotAllowedException
-
-                _configuredConformanceProvider
-                    .ConfigureOptionalCapabilities(x =>
+            // Turns on conditional creates, even when not configured in the DefaultCapabilities.json
+            // If this is not enabled in the capability statement
+            // ConditionalCreateResourceRequest will fail with a MethodNotAllowedException
+            _configuredConformanceProvider
+                .ConfigureOptionalCapabilities(x =>
+                {
+                    foreach (ListedResourceComponent r in x.Rest.Server().Resource)
                     {
-                        foreach (ListedResourceComponent r in x.Rest.Server().Resource)
+                        if (r.Interaction.Any(y => string.Equals(y.Code, TypeRestfulInteraction.Create, StringComparison.Ordinal)))
```

Comment:
I'd probably just incorporate this into CapabilityStatementBuilder now

Reply:
I was looking where to put it, thanks for pointing out!